pkg/cmd/server/api/helpers.go

```diff
@@ -305,6 +305,7 @@ func GetKubeClient(kubeConfigFile string) (*kclient.Client, *restclient.Config,
 	// TODO: this should be configured by the caller, not in this method.
 	kubeConfig.QPS = 100.0
 	kubeConfig.Burst = 200
+	kubeConfig.ContentType = "application/vnd.kubernetes.protobuf"
```

Comment:
These are going to be config flags.

On Fri, Jul 22, 2016 at 1:15 PM, David Eads <notifications@github.com>
wrote:

> In pkg/cmd/server/api/helpers.go
> <https://github.com/openshift/origin/pull/9814#discussion_r71913042>:
>
> > @@ -305,6 +305,7 @@ func GetKubeClient(kubeConfigFile string) (*kclient.Client, *restclient.Config,
> >  	// TODO: this should be configured by the caller, not in this method.
> >  	kubeConfig.QPS = 100.0
> >  	kubeConfig.Burst = 200
> > +	kubeConfig.ContentType = "application/vnd.kubernetes.protobuf"
>
> This needs to fallback to json to handle a rolling upgrade case. These
> functions are used when wiring up controllers, so a new controller to old
> master case needs to send and accept json.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/9814/files/ef32c3c969d72c7efec9d4ee377d4d43d290d761..895443fc49722a65e8fb36a0e28540ae6f260b7a#r71913042>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABG_p5HKNRa7RWR6RNSj1FnuXTVH2m1cks5qYPq2gaJpZM4JLFc7>
> .
>